src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs

```diff
@@ -2158,6 +2158,123 @@ static void Test2()
                 );
         }
 
+        [Fact]
+        public void ImplementsIEnumerableT_22_ObsoleteConstructor()
+        {
+            var src = """
+using System.Collections;
+using System.Collections.Generic;
+
+class MyCollection : IEnumerable<long>
+{
+    [System.Obsolete()]
+    public MyCollection(){}
+    public List<long> Array = new List<long>();
+    IEnumerator<long> IEnumerable<long>.GetEnumerator() => throw null;
+    IEnumerator IEnumerable.GetEnumerator() => throw null;
+
+    public void Add(long l) => Array.Add(l);
+}
+
+class Program
+{
+    static void Main()
+    {
+#line 100
```

Comment:
Is use of `#line` relevant to the test?

Reply:
> Is use of `#line` relevant to the test?

I used it intentionally, it is reflected in the line numbers for diagnostics in the base line